src/jvm/benchmark/org/jetbrains/kotlinx/lincheck_benchmark/AbstractLincheckBenchmark.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
+package org.jetbrains.kotlinx.lincheck_benchmark
+
+import org.jetbrains.kotlinx.lincheck.*
+import org.jetbrains.kotlinx.lincheck.strategy.*
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.jetbrains.kotlinx.lincheck.strategy.stress.StressOptions
+import kotlin.reflect.KClass
+import org.junit.Test
+
+
+abstract class AbstractLincheckBenchmark(
+    private vararg val expectedFailures: KClass<out LincheckFailure>
+) {
+
+    @Test(timeout = TIMEOUT)
```

Comment:
Why not use JMH to collect the statistics and generate a CSV file?

Reply:
When started working on benchmarks, I actually considered JMH or kotlinx-benchmarks at some point.

Why I have not used it in the end?
Short answer: JMH is primarly for doing micro-benchmarking of individual functions or data structures,
while the benchmarks from this PR are macro-benchmarks for the framework as a whole.

Long answer.

Benchmarks from this PR are "integration" benchmarks, we evaluate the performance of the framework as a whole. We run the whole Lincheck pipeline: scenario generator, instrumentation, stress or model checking strategy, verifier, etc. We write these benchmarks similarly how a user would write a Lincheck tests: using `StressOptions` or `ModelCheckingOptions`, setting up all the parameters of the tests, etc.

In contrast, JMH would take some benchmark method annotated with `@Benchmark`, then it would generate code to run it multiple times according to its own algorithm, calculate various statistics, etc.
This would be significantly different from what we want from benchmarks in this PR.

Also, we want to have some control over what statistics we collect in our benchmarks.
For example, with current implementation in this PR we can track per-iteration or per-invocation statistics,
and we only measure the running time of the code under Lincheck testing.
In the feature, we may also want to further refine the statistics, e.g. by splitting the time measurements to actual test running time, time taken by verifier, etc.
And we also might want to collect other metrics beyond execution time, e.g. coverage.

Reply:
Thanks for the explanation!